Doppler.Sap/Mappers/Billing/BillingForArMapper.cs

```diff
@@ -135,53 +141,94 @@ public SapSaleOrderModel MapDopplerUpdateBillingRequestToSapSaleOrder(UpdatePaym
         }
 
 
-        public SapCreditNoteModel MapToSapCreditNote(SapSaleOrderInvoiceResponse sapSaleOrderInvoiceResponse, int creditNoteType)
+        public SapCreditNoteModel MapToSapCreditNote(SapSaleOrderInvoiceResponse sapSaleOrderInvoiceResponse, CreditNoteRequest creditNoteRequest)
         {
             var sapCreditNoteModel = new SapCreditNoteModel
             {
                 NumAtCard = sapSaleOrderInvoiceResponse.NumAtCard ?? "",
-                U_DPL_INV_ID = sapSaleOrderInvoiceResponse.U_DPL_INV_ID,
+                U_DPL_CN_ID = creditNoteRequest.CreditNoteId,
                 DocumentLines = new List<SapCreditNoteDocumentLineModel>(),
                 DocDate = _dateTimeProvider.GetDateByTimezoneId(_dateTimeProvider.UtcNow, _timezoneConfig.InvoicesTimeZone).ToString("yyyy-MM-dd"),
                 DocDueDate = _dateTimeProvider.GetDateByTimezoneId(_dateTimeProvider.UtcNow, _timezoneConfig.InvoicesTimeZone).ToString("yyyy-MM-dd"),
                 TaxDate = _dateTimeProvider.GetDateByTimezoneId(_dateTimeProvider.UtcNow, _timezoneConfig.InvoicesTimeZone).ToString("yyyy-MM-dd"),
-                InvoiceId = sapSaleOrderInvoiceResponse.U_DPL_INV_ID
+                CreditNoteId = creditNoteRequest.CreditNoteId,
+                CardCode = sapSaleOrderInvoiceResponse.CardCode,
+                TransferReference = creditNoteRequest.TransferReference
             };
 
-            var creditNoteData = (creditNoteType == (int)CreditNoteEnum.NoRefund)
-                ? new
-                {
-                    FreeText = $"Cancela la factura: {sapSaleOrderInvoiceResponse.DocNum}."
-                }
-                : new
-                {
-                    FreeText = string.Empty
-                };
+            var balance = creditNoteRequest.Amount;
+            var isPartial = decimal.ToDouble(sapSaleOrderInvoiceResponse.DocTotal) > (creditNoteRequest.Amount);
```

Comment:
Why do you need to convert it to double? it is not better to convert to decimal?

Reply:
For nothing in special, I can change it to convert to decimal.

Reply:
Done